### PR TITLE
Multipart android

### PR DESCRIPTION
**Pull request by doug-stritt**
Story: https://app.shortcut.com/wellth-inc/story/27495/action-rewards-multipart-photo-upload-on-android

Fix multipart uploads for action rewards

Screenshot 2024-02-28 at 8 29 54 PM

### CODE REVIEW

**Review by heywats — APPROVED** [review]

File: android/src/main/java/com/appfolio/uploader/UploaderModule.kt [hunk 1]
```diff
@@ -200,16 +202,26 @@ class UploaderModule(val reactContext: ReactApplicationContext) : ReactContextBa
         ModifiedBinaryUploadRequest(this.reactApplicationContext, url!!, limitNetwork)
                 .setFileToUpload(filePath!!)
       } else {
-        if (!options.hasKey("field")) {
-          promise.reject(java.lang.IllegalArgumentException("field is required field for multipart type."))
-          return
+        if (parts != null && parts.size() < 1) {
```
Comment (kyleweisel) [inline android/src/main/java/com/appfolio/uploader/UploaderModule.kt:205]: Should this be `parts == null || parts.size() < 1`?

**Review by kyleweisel — COMMENTED** [review]
I think this is good, just a quick comment/question.

File: android/src/main/java/com/appfolio/uploader/UploaderModule.kt (hunk 1 above)
Comment (doug-stritt) [reply android/src/main/java/com/appfolio/uploader/UploaderModule.kt:205]: Oh yea either way. The null check is so I don't get an exception when checking parts.size() but there's a null check in the app before it gets here to protect against it.

Comment (jad-at-wellth) [reply android/src/main/java/com/appfolio/uploader/UploaderModule.kt:205]: @kyleweisel is correct. Change it to his line.

Comment (doug-stritt) [reply android/src/main/java/com/appfolio/uploader/UploaderModule.kt:205]: I updated it

**Pull request merged**